### DISCUSSION TITLE
Missing index on elements table causes slow DB performance

### DISCUSSION THREAD

**Discussion by lenvanessen** [craftcms/cms#10418] [Ideas] [answered]
Hi There,

We found after some optimizations and research that the `elements` table has a multi-column index containing:
`archived,dateDeleted,draftId,revisionId,canonicalId`.

However, many queries Craft executes also include the column `enabled`, causing the index to not actually have an effect on the performance. After we added a new index that contained the column, see what happend to our overal load on our master mysql server, it dropped by +-60%:
![Schermafbeelding 2022-01-27 om 10 42 47](https://user-images.githubusercontent.com/23024452/151334687-542cacf9-bd84-4af0-8ab0-8ef4de30a173.png)

I believe it could be valuable to add the index by default, especially for Craft sites that have many rows in the elements table

**Comment by brandonkelly** [comment]
Interesting! Mind sharing the exact index columns you went with?

**Reply by lenvanessen** [reply]
![Schermafbeelding 2022-02-09 om 09 07 37](https://user-images.githubusercontent.com/23024452/153151785-7002407f-1e4b-4e63-aa07-14c62ff791d4.png)

**Reply by brandonkelly** [reply]
Thanks! Added for the next release.

**Comment by brandonkelly** [comment, marked as answer]
Craft 3.7.32 is out now with this change.
